src/utilities/Viewport/index.jsx

```diff
@@ -0,0 +1,54 @@
+/* global window */
+/* eslint no-console: off */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import throttle from 'lodash/throttle';
+import { getConfiguration, ScreenContext } from '../../config';
+import { getScreenClass as getDefaultScreenClass } from '../../utils';
+import ResizeObserver from 'resize-observer-polyfill'
```

Comment:
do we really need an extra dependency if we can just use `window.innerWidth`?

Reply:
The key point of this component is that we use a parent element's size as if it was window.innerWidth, which provides the ability to subdivide the screen.